src/transformers/models/mamba/modeling_mamba.py

```diff
@@ -563,25 +620,36 @@ def forward(
         if self.gradient_checkpointing and self.training and use_cache:
             use_cache = False
 
-        if cache_params is None and use_cache:
-            cache_params = MambaCache(
-                self.config, inputs_embeds.size(0), device=inputs_embeds.device, dtype=inputs_embeds.dtype
-            )
+        if use_cache:
+            if cache_params is None:
+                cache_params = MambaCache(
+                    self.config, inputs_embeds.size(0), device=inputs_embeds.device, dtype=inputs_embeds.dtype
+                )
+                cache_position = torch.arange(0, self.config.conv_kernel, device=inputs_embeds.device)
+            elif cache_position is None:
+                # cases when we do manual forward instead of using `model.generate`
+                # throw error instead of doing some hack to conjecture the current cache position
+                raise RuntimeError(
```

Comment:
Cool, this is the single breaking change, as users who used to pass cache_params manually will now have to pass the cache_positions. We could add a not to mention that generate does that automatically

Reply:
Yes, I think users using manual forward and cache should have a clear idea of what they are trying to do by passing correct arguments